### PR TITLE
Implement storage-network

### PR DESCRIPTION
**Pull request by c3y1huang**
https://github.com/longhorn/longhorn/issues/2285
https://github.com/longhorn/longhorn/issues/3546

### CODE REVIEW

File: pkg/datasource/handler.go [hunk 1]
```diff
@@ -283,6 +286,14 @@ func (s *Service) exportFromVolume(parameters map[string]string) error {
 		return fmt.Errorf("can't get pod ip from environment variable")
 	}
 
+	storageIP := podIP
+	if ip, err := s.getLocalIPv4fromInterface(StorageNetworkInterface); err != nil {
+		s.log.WithError(err).Debugf("Failed to get IP from %v interface", StorageNetworkInterface)
```
Comment (shuo-wu) [inline pkg/datasource/handler.go:291]: NIT: "..., fallback to use the default pod IP %v"

**Review by shuo-wu — APPROVED** [review]

File: pkg/datasource/handler.go (hunk 1 above)
Comment (c3y1huang) [reply pkg/datasource/handler.go]: Updated.

**Review by shuo-wu — APPROVED** [review]
LGTM

File: pkg/datasource/handler.go [hunk 2]
```diff
@@ -26,6 +27,8 @@ import (
 const (
 	RetryInterval = 1 * time.Second
 	RetryCount    = 60
+
+	StorageNetworkInterface = "lhnet1"
```
Comment (innobead) [inline pkg/datasource/handler.go:31]: How do know the network interface name here?

Comment (c3y1huang) [reply pkg/datasource/handler.go:31]: It's set by Longhorn manager.
```
StorageNetworkInterface = "lhnet1"
```

Comment (innobead) [reply pkg/datasource/handler.go:31]: ref: https://github.com/longhorn/longhorn-manager/pull/1212/files#diff-f8d2bd35e1a8c9b86c8c4448dde00bc5480652f85ed0630efc012c3906d3a14fR134

**Review by innobead — APPROVED** [review]

**Pull request merged**